src/lib.rs

```diff
@@ -590,12 +596,23 @@ impl KafkaJsonToDelta {
                 .last_transaction_version(&self.app_id_for_partition(*partition));
 
             if let Some(version) = version {
-                if *offset != version {
-                    info!(
-                        "Conflict offset for partition {}: state={:?}, delta={:?}",
-                        partition, offset, version
-                    );
-                    result = false;
+                match offset {
+                    Some(offset) => {
+                        if *offset != version {
+                            info!(
+                                "Conflict offset for partition {}: state={:?}, delta={:?}",
+                                partition, offset, version
+                            );
+                            result = false;
+                        }
+                    }
+                    None => {
+                        info!(
+                            "Conflict offset for partition {}: state=<none>, delta={:?}",
+                            partition, version
+                        );
+                        result = false;
+                    }
```

Comment:
What would you think about a noop to shorten this up? something like

```
                    Some(offset) if *offset == version => { /* noop - partition offsets match */ }
                    _ => {
                        info!(
                            "Conflict offset for partition {}: state={:?}, delta={:?}",
                            partition, offset, version
                        );
                        result = false;
                    }

```

Reply:
that's a good one, will check, just need to figure out logs